js/speedtest/speedtest.js

```diff
@@ -0,0 +1,374 @@
+(function ($) {
+  function Speedtest() {
+    this._serverList = [];
+    this._selectedServer = null;
+    this._settings = {};
+    this._state = 0;
+  }
+  Speedtest.prototype = {
+    constructor: Speedtest,
+    getState: function() {
+      return this._state;
+    },
+	  setParameter: function(parameter, value) {
+	    if (this._state == 3) {
+		    throw "You cannot change the test settings while running the test";
+      }
+	    this._settings[parameter] = value;
+	    if (parameter !== "telemetry_extra") {
+        return;
+	    }
+      this._originalExtra = this._settings.telemetry_extra;
+	  },
+	  _checkServerDefinition: function(server) {
+	    try {
+		    if (typeof server.name !== "string")
+		      throw "Name string missing from server definition (name)";
+		    if (typeof server.server !== "string")
+		      throw "Server address string missing from server definition (server)";
+		    if (server.server.charAt(server.server.length - 1) != "/")
+		      server.server += "/";
+		    if (server.server.indexOf("//") == 0)
+		      server.server = location.protocol + server.server;
+		    if (typeof server.dlURL !== "string")
+		      throw "Download URL string missing from server definition (dlURL)";
+		    if (typeof server.ulURL !== "string")
+		      throw "Upload URL string missing from server definition (ulURL)";
+		    if (typeof server.pingURL !== "string")
+		      throw "Ping URL string missing from server definition (pingURL)";
+		    if (typeof server.getIpURL !== "string")
+		      throw "GetIP URL string missing from server definition (getIpURL)";
+	      } catch (e) {
+		      throw "Invalid server definition";
+	      }
+	  },
+	  addTestPoint: function(server) {
+	    this._checkServerDefinition(server);
+	    if (this._state == 0) {
+        this._state = 1;
+      }
```

Comment:
indentation + unclear variable name 'state' who should just be the length of the serve list ( state is irrelevant )